### ISSUE TITLE
[Feature request] Extending limit of the canvas/background

### ISSUE THREAD

**Issue by barbershop9** [juzzlin/Heimer#126]
First of all, awesome work on this project (to everyone involved).

I've been using this for a couple times now (two weeks) and after extensive uses, a glaring problem occurred to me:
the canvas/background was limited in size.

By that, i do mean this:

![2020-12-21_19-09](https://user-images.githubusercontent.com/76448436/102808354-82aba180-43c0-11eb-9daa-109ef28a9768.png)

This is even more noticeable on a "decently" sized map:

![2020-12-21_19-11](https://user-images.githubusercontent.com/76448436/102808426-97883500-43c0-11eb-8e6a-447ecc68de26.png)

I didn't found any way to extend it, and I'm not really knowledgeable in C or QT...
I did try to make a node in the space shown in the last pic (after the red lines), but of course, it couldn't be accessed by zooming in or out, since the red part prevented to view it.

I'm aware i could just split my "thought" or map into different files, but that wouldn't feel "right" to me, and managing multiple map of the same subject or thought would feel clunky at some time, but that's maybe just me...

I wouldn't ask for an "infinite canvas", since that may be harder to implement or out of scope for this project, but having at least a way to change the limit of canvas, so one wouldn't be limited in their writing or "thoughts", that would be nice to have!

PS: I'm using this on windows, though i have linux installed.

**Comment by juzzlin** [comment]
Yeah, this is something that needs to be improved. I'll figure out something. Thanks for reporting!

**Comment by barbershop9** [comment]
No problem. Also, this is the fastest response time i seen on github issues :+1:

**Comment by barbershop9** [comment]
If it's too hard to implement, maybe i could just change the value that is hard-coded in the source files, to increase the default limit of the canvas, and compile to see if it worked?

just need to know where it is, not sure where it is declared

**Comment by juzzlin** [comment]
I'm glad to hear you like the simplicity of Heimer :) That was exactly why I decided to create it.

The scene size is defined as `Constants::Scene::RADIUS` in `constants.hpp` and that's used in `editor_scene.cpp`. I'd like to have an unlimited area myself. Didn't have time to investigate yet, but I guess the current grid implementation could make it tricky. I have noticed that the area is maybe a bit too tight so I'll definitely do something to this anyway.
